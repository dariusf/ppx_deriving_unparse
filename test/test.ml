type expr =
  | Lit of int [@form _1]
  | Plus of expr * expr [@form _1 "+" _2] [@prec left, 3]
  | Times of expr * expr [@form _1 "*" _2] [@prec left, 4]
[@@deriving unparse]

let () = print_endline (render_expr (Times (Plus (Lit 1, Lit 2), Lit 3)))
