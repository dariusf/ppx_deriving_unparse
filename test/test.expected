(1 + 2) * 3
a-1->b: c1; (a-1->b: c1 * a-1->b: c1); end
(λ.0) (λ.0)
λ.0 (λ.0)
aω ∨ ((⊥⋅c)∞ ∨ (ε⋅b)✭⋅b✭)
